Windows/CommonDialog/MainWindow.xaml.cs

```diff
@@ -138,29 +136,30 @@ private void Button_OpenFolderDialog(object sender, RoutedEventArgs e)
 
             //openFolderDialog.Multiselect = true;
 
-            //string text = "";
+            //StringBuilder sb = new StringBuilder();
             //if (openFolderDialog.ShowDialog() == true)
             //{
-            //    text = "List of all subdirectories in the selected folders : \n\n";
+            //    sb.AppendLine($"List of all subdirectories in the selected folders :");
+            //    sb.AppendLine();
+
             //    foreach (var folder in openFolderDialog.FolderNames)
             //    {
-            //        DirectoryInfo d = new DirectoryInfo(folder);
-            //        text += folder + "\n---\n";
+            //        sb.AppendLine($"Folder Name : {folder}");
 
-            //        d.GetDirectories().ToList().ForEach(x => text += x.FullName + "\n");
+            //        DirectoryInfo d = new DirectoryInfo(folder);
+            //        d.GetDirectories().ToList().ForEach(x => sb.AppendLine($"\t{x.Name}"));
```

Comment:
So what you could do is
```c#
foreach (var folder in openFolderDialog.FolderNames)
{
   sb.AppendLine($"Folder Name : {folder}");
   foreach (var subfolder in Directory.EnumerateDirectories(folder))
   {
      sb.AppendLine($"\t{subfolder}");
   }
}
```

instead of creating a copy of the array just to call ForEach on it